kustomize/commands/localize/localize.go

```diff
@@ -27,7 +25,6 @@ type flags struct {
 
 // NewCmdLocalize returns a new localize command.
 func NewCmdLocalize(fs filesys.FileSystem, writer io.Writer) *cobra.Command {
-	log.SetOutput(writer)
```

Comment:
This definitely needs to be removed (and we should have a CI check banning it outside of tests, if we can), but I wonder if localize is somewhere expecting to be able to write to stdout using `log`. @annasong20 can you provide context?

Reply:
Somewhat related but not a cause of the bug: the success message at L66 in this file should be output to stderr, not stdout.

Reply:
I can add that in

Reply:
Should it be an error, or a log? Returning an error populated with success message feels off to me.

As in should I log it to standard error and then return nil?

Reply:
IMO it should be logged, but to the stderr stream. A long time ago, someone shared this explanation with me (the original website seems gone now, but this looks like the same content): https://www.pixelstech.net/article/1326560863-When-to-use-STDERR-instead-of-STDOUT.

Essentially, instead of thinking out=messages and err=errors, it is more like out=data and err=diagnostics. For `kustomize build`, this is particularly clear: only the build results must go to stdout, and everything else necessarily needs to go to stderr (or the output will not be parseable! That's why this regression is kinda bad!). Localize output is on disk, so it probably doesn't have any 'default' output on stdout.

Reply:
No, localize does not need to write to stdout anywhere else. The only localize printed messages other than this success are error messages that can go to stderr. 

I'm sorry for this regression and all the time @cailynse you spent debugging it. I forgot this code was always executed and regardless, shouldn't have used log.SetOutput anyway. Thank you so much for fixing it.

Reply:
Ain't no thang @annasong20 - I likely would have done the same thing and I learned a lot while debugging!